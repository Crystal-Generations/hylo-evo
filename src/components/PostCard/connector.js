import { connect } from 'react-redux'
import { fetchPost } from 'store/actions/posts'
import { getPostById } from 'store/selectors/getPostById'

export function mapStateToProps (state, props) {
  return {
    post: getPostById(state, props)
  }
}

export const mapDispatchToProps = { fetchPost }

export default connect(mapStateToProps, mapDispatchToProps)
