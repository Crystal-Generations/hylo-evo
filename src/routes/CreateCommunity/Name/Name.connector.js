import { connect } from 'react-redux'
import { push } from 'react-router-redux'

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-community/domain'))
  }
}

export default connect(null, mapDispatchToProps)
