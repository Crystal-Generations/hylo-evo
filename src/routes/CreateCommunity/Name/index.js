import React, { Component } from 'react'
import '../CreateCommunity.scss'
import LeftSidebar from '../../Signup/LeftSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from '../ModalFooter'

const theme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}
export default class Name extends Component {
  setState = () => {

  }
  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        header="Great, let's get started"
        body="All good things start somewhere! Let's kick things off with a catchy name for your community."
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 1/4</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <span styleName='text-input-label'>What's the name of your community?</span>
          <TextInput
            type='text'
            name='community-name'
            onChange={this.setState('community-name')}
            inputRef={input => { this.email = input }}
            theme={theme}
            placeholder="What's the name of your community?"
          />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        previous={this.previous}
        showPrevious={false}
        continueText={'Continue'}
        />
    </div>
  }
}
