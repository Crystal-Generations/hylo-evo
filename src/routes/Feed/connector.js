import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'
import { fetchPosts } from './actions'

function mapStateToProps (state) {
  return {
    feedItems: state.posts
  }
}

export const mapDispatchToProps = { fetchPosts }

export default connect(mapStateToProps, mapDispatchToProps)
