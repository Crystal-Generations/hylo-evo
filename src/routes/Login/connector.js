import { connect } from 'react-redux'
import { checkLogin, login, pickError } from './store'

export default connect(pickError, {checkLogin, login})
