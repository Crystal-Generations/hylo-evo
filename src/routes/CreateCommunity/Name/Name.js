import React, { Component } from 'react'
import '../CreateCommunity.scss'
import LeftSidebar from '../../Signup/LeftSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from '../ModalFooter'

export default class Name extends Component {
  constructor (props) {
    super(props)
    this.state = {
      communityName: ''
    }
  }

  handleNameChange = (event) => {
    const communityName = event.target.value
    this.setState({
      communityName
    })
  }

  submit = () => {
    this.props.addCommunityName(this.state.communityName)
    this.props.goToNextStep()
  }

  onEnter = event => {
    if (event.key === 'Enter') {
      this.submit()
    }
  }

  componentWillMount = () => {
    const { communityName } = this.props
    if (communityName) this.setState({name: communityName})
  }

  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        theme={sidebarTheme}
        header="Great, let's get started"
        body="All good things start somewhere! Let's kick things off with a catchy name for your community."
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 1/3</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <span styleName='text-input-label'>{ this.state.communityName && "What's the name of your community?"}</span>
          <TextInput
            type='text'
            name='community-name'
            onChange={this.handleNameChange}
            value={this.state.communityName}
            theme={inputTheme}
            placeholder="What's the name of your community?"
            showClearButton={false}
            onEnter={this.onEnter}
          />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        showPrevious={false}
        continueText={'Continue'}
        />
    </div>
  }
}

const inputTheme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}
