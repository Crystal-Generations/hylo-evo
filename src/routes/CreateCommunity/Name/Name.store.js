import { ADD_COMMUNITY_NAME } from '../CreateCommunity.store'

export function addName (name) {
  return {
    type: ADD_COMMUNITY_NAME,
    payload: {
      name
    }
  }
}
