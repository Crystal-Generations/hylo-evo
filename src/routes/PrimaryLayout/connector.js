import { connect } from 'react-redux'
import { SAMPLE_COMMUNITY } from 'routes/Feed/sampleData'

const SAMPLE_USER = {
  id: '1',
  firstName: 'Axolotl',
  lastName: 'Jones',
  avatarUrl: 'https://d3ngex8q79bk55.cloudfront.net/user/13986/avatar/1444260480878_AxolotlPic.png'
}

function mapStateToProps (state) {
  return {
    community: SAMPLE_COMMUNITY,
    currentUser: SAMPLE_USER
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
