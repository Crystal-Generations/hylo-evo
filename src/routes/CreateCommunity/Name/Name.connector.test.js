import { mapDispatchToProps } from './Name.connector'
describe('Domain', () => {
  it('should call fetchCommunity from mapDispatchToProps', () => {
    const dispatch = jest.fn(x => x)
    const props = {}
    const communityName = 'Community Name'
    const dispatchProps = mapDispatchToProps(dispatch, props)
    expect(dispatchProps.addCommunityName(communityName)).toMatchSnapshot()
  })
})
