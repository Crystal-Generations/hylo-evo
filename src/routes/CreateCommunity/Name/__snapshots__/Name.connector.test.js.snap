// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Domain should call fetchCommunity from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "name": "Community Name",
  },
  "type": "CREATE_COMMUNITY_ADD_COMMUNITY_NAME",
}
`;

exports[`Domain should call goToNextStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-community/domain",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
